parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestEncryption.java

```diff
@@ -0,0 +1,215 @@
+        .withKeyID("ck0")
+        .build();
+    columnProperties1 = ColumnEncryptionProperties.builder("int32_field")
+        .withKey(columnKey1)
+        .withKeyID("ck1")
+        .build();
+    columnPropertiesMap = new HashMap<ColumnPath, ColumnEncryptionProperties>();
+    columnPropertiesMap.put(columnProperties0.getPath(), columnProperties0);
+    columnPropertiesMap.put(columnProperties1.getPath(), columnProperties1);
+    encryptionProperties = FileEncryptionProperties.builder(footerKey)
+        .withAlgorithm(ParquetCipher.AES_GCM_CTR_V1)
+        .withFooterKeyID("fk")
+        .withAADPrefix(AADPrefix)
+        .withEncryptedColumns(columnPropertiesMap)
+        .build();
+    encryptionPropertiesList[3] = encryptionProperties;
+    decryptionPropertiesList[3] = decryptionProperties; // Same decryption properties
+
+    // #4  Plaintext footer, default algorithm, key metadata, key retriever, AAD
+    columnProperties0 = ColumnEncryptionProperties.builder("binary_field")
+        .withKey(columnKey0)
+        .withKeyID("ck0")
+        .build();
+    columnProperties1 = ColumnEncryptionProperties.builder("int32_field")
+        .withKey(columnKey1)
+        .withKeyID("ck1")
+        .build();
+    columnPropertiesMap = new HashMap<ColumnPath, ColumnEncryptionProperties>();
+    columnPropertiesMap.put(columnProperties0.getPath(), columnProperties0);
+    columnPropertiesMap.put(columnProperties1.getPath(), columnProperties1);
+    encryptionProperties = FileEncryptionProperties.builder(footerKey)
+        .withFooterKeyID("fk")
+        .withPlaintextFooter()
+        .withAADPrefix(AADPrefix)
+        .withEncryptedColumns(columnPropertiesMap)
+        .build();
+    encryptionPropertiesList[4] = encryptionProperties;
+    decryptionPropertiesList[4] = decryptionProperties; // Same decryption properties
+
+
+    MessageType schema = parseMessageType(
+        "message test { "
+            + "required binary binary_field; "
+            + "required int32 int32_field; "
+            + "required int64 int64_field; "
+            + "required boolean boolean_field; "
+            + "required float float_field; "
+            + "required double double_field; "
+            + "required fixed_len_byte_array(3) flba_field; "
+            + "required int96 int96_field; "
+            + "} ");
+    GroupWriteSupport.setSchema(schema, conf);
+    SimpleGroupFactory f = new SimpleGroupFactory(schema);
+
+    for (int encryptionMode = 0; encryptionMode < numberOfEncryptionModes; encryptionMode++) {
+      System.out.println("MODE: "+encryptionMode);
+      
+      Path file = new Path(root, "m_" + encryptionMode + ".parquet.encrypted");
+      ParquetWriter<Group> writer = new ParquetWriter<Group>(
+          file,
+          new GroupWriteSupport(),
+          UNCOMPRESSED, 1024, 1024, 512, true, false, ParquetWriter.DEFAULT_WRITER_VERSION, conf, 
+          encryptionPropertiesList[encryptionMode]);
+      for (int i = 0; i < 1000; i++) {
```

Comment:
I don't know if make to much sense to write the same data 1000 times. Most of our tests are working by generating random data in memory (e.g. a `List<Group>`) the write it and then test whether we can read back the same data we have in memory.

Reply:
we're working on a better unitest for encryption, in #782 . it should be possible to drop this one.